tools/clang/test/HLSLFileCheck/hlsl/functions/return_type/array/lib_function.hlsl

```diff
@@ -0,0 +1,18 @@
+// RUN: %dxc -T lib_6_3 %s | FileCheck %s
+// RUN: %dxc -T lib_6_6 %s | FileCheck %s
+
+// Tests that non-entry point functions can return arrays in library profile.
+
+// CHECK: define void
+// CHECK-SAME: getA
+// CHECK-SAME: ([2 x i32]* noalias nocapture sret %[[result:.*]])
+export int getA() [2]
+{
+  // CHECK: %[[gep1:.*]] = getelementptr inbounds [2 x i32], [2 x i32]* %[[result]], i32 0, i32 0
+  // CHECK: store i32 1, i32* %[[gep1]]
+  // CHECK: %[[gep2:.*]] = getelementptr inbounds [2 x i32], [2 x i32]* %[[result]], i32 0, i32 1
+  // CHECK: store i32 2, i32* %[[gep2]]
+  int a[2] = { 1, 2 };
+  return a;
+}
```

Comment:
Not essential, but another way to encounter this would be with an out param that gets assigned wholesale from a local variable. Something like 
```hlsl
export int getB(out uint b[2]) {
  int a[2] = {1,2};
  b = a;
}
```

Reply:
Yup!  But it would need some slight fixes there (void return type, match out type with local):
```
export void getB(out int b[2]) {
  int a[2] = {1,2};
  b = a;
}
```

Interestingly, if b doesn't match a in component type, I think HLSL should be able to implicitly convert, but instead we get this error:
```
error: cannot implicitly convert from 'int [2]' to 'uint __restrict[2]'
  b = a;
    ^
```
This looks like another bug 😒

At least it works with a cast:
```
  b = (uint[2])a;
```

Reply:
Filed #4458 for the above issue.